File: Client/src/app/models/ProductStockDTO.ts
import { WalmartProductDTO } from "./WalmartProductDTO";

export class ProductStockDTO {
  id?: number = undefined;
  name?: string = undefined;
  units?: number = undefined;
  unitType?: number = undefined;
  productId?: number | undefined = undefined;
  walmartProduct?: WalmartProductDTO = undefined;
}

File: API/ContainerNinja.Contracts/DTO/ProductStockDTO.cs

using ContainerNinja.Contracts.Enum;

namespace ContainerNinja.Contracts.DTO
{
    public class ProductStockDTO
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public float? Units { get; set; }

        public UnitType UnitType { get; set; }

        public int? ProductId { get; set; }

        public WalmartProductDTO Product { get; set; }
    }
}